frontend/src/components/RoadmapSelectorWidget.tsx

```diff
@@ -62,15 +62,23 @@ export const RoadmapSelectorWidget = () => {
 
   return (
     <Dropdown css={css} title={selectedRoadmap} maxLength={21}>
-      {roadmaps.map((roadmap) => (
-        <Link
-          key={roadmap.id}
-          className={classes(css.dropItem)}
-          to={`${paths.roadmapHome}/${roadmap.id}${paths.roadmapRelative.dashboard}`}
-        >
-          {roadmap.name}
-        </Link>
-      ))}
+      {roadmaps.map((roadmap) => {
+        const roadmapPath = `${paths.roadmapHome}/${roadmap.id}${paths.roadmapRelative.dashboard}`;
+        return (
+          <Link
+            key={roadmap.id}
+            className={classes(css.dropItem)}
+            onClick={(e) => {
+              e.preventDefault();
+              dispatch(roadmapsActions.clearTaskmapPosition());
+              history.push(roadmapPath);
+            }}
+            to={roadmapPath}
+          >
+            {roadmap.name}
+          </Link>
+        );
+      })}
```

Comment:
```suggestion
      {roadmaps.map((roadmap) => (
        <Link
          key={roadmap.id}
          className={classes(css.dropItem)}
          onClick={() => {
            dispatch(roadmapsActions.clearTaskmapPosition());
          }}
          to={`${paths.roadmapHome}/${roadmap.id}${paths.roadmapRelative.dashboard}`}
        >
          {roadmap.name}
        </Link>
      ))}
```

The click handler could be used for just dispatching the redux action, the default would then follow the link

Reply:
I thought I tried this with the action not dispatching. Looks like it actually works perfectly fine: thank you